actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -852,7 +858,7 @@ def each_element(object)
         when Parameters
           if object.fields_for_style?
             hash = object.class.new
-            object.each { |k, v| hash[k] = yield v }
+            object.each { |k, v| hash[k] = yield v if Parameters.field_style_valid?(k, v) }
```

Comment:
I wonder if adding a `#each_nested`, `#each_field` or something like that that does the filtering for you is worth it.